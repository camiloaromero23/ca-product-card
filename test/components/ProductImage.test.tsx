import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';

import { product2 } from '../data/products';

describe('ProductImage', () => {
  it('should render the component correctly with its custom image', () => {
    const wrapper = renderer.create(<ProductImage img="https://hello.jpg" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should show the component with its product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
