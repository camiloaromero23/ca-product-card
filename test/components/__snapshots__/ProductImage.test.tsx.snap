// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage should render the component correctly with its custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://hello.jpg"
/>
`;

exports[`ProductImage should show the component with its product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
